cluster-autoscaler/cloudprovider/gce/autoscaling_gke_client_v1beta1.go

```diff
@@ -0,0 +1,193 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gce
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/autoscaler/cluster-autoscaler/cloudprovider"
+	"k8s.io/autoscaler/cluster-autoscaler/utils/gpu"
+	"k8s.io/autoscaler/cluster-autoscaler/utils/units"
+
+	apiv1 "k8s.io/api/core/v1"
+
+	"github.com/golang/glog"
+	gke_beta "google.golang.org/api/container/v1beta1"
+)
+
+var (
+	// This makes me so sad
+	taintEffectsMap = map[apiv1.TaintEffect]string{
+		apiv1.TaintEffectNoSchedule:       "NO_SCHEDULE",
+		apiv1.TaintEffectPreferNoSchedule: "PREFER_NO_SCHEDULE",
+		apiv1.TaintEffectNoExecute:        "NO_EXECUTE",
+	}
+)
+
+func (m *autoscalingGkeClientV1beta1) FetchNodePools() ([]NodePool, error) {
+	nodePoolsResponse, err := m.gkeBetaService.Projects.Locations.Clusters.NodePools.List(m.clusterPath).Do()
+	if err != nil {
+		return nil, err
+	}
+	nodePools := []NodePool{}
+	for _, pool := range nodePoolsResponse.NodePools {
+		if pool.Autoscaling != nil {
+			nodePools = append(nodePools, NodePool{
+				Name:              pool.Name,
+				InstanceGroupUrls: pool.InstanceGroupUrls,
+				Autoscaled:        pool.Autoscaling.Enabled,
+				MinNodeCount:      pool.Autoscaling.MinNodeCount,
+				MaxNodeCount:      pool.Autoscaling.MaxNodeCount,
+				Autoprovisioned:   pool.Autoscaling.Autoprovisioned,
+			})
+		}
+	}
+	return nodePools, nil
+}
+
+func (m *autoscalingGkeClientV1beta1) FetchLocations() ([]string, error) {
+	cluster, err := m.gkeBetaService.Projects.Locations.Clusters.Get(m.clusterPath).Do()
+	return cluster.Locations, err
+}
+
+func (m *autoscalingGkeClientV1beta1) FetchResourceLimits() (*cloudprovider.ResourceLimiter, error) {
+	cluster, err := m.gkeBetaService.Projects.Locations.Clusters.Get(m.clusterPath).Do()
+	if err != nil {
+		return nil, err
+	}
+	if cluster.Autoscaling == nil {
+		return nil, nil
+	}
+
+	minLimits := make(map[string]int64)
+	maxLimits := make(map[string]int64)
+	for _, limit := range cluster.Autoscaling.ResourceLimits {
+		if _, found := supportedResources[limit.ResourceType]; !found {
+			glog.Warningf("Unsupported limit defined %s: %d - %d", limit.ResourceType, limit.Minimum, limit.Maximum)
+		}
+		minLimits[limit.ResourceType] = limit.Minimum
+		maxLimits[limit.ResourceType] = limit.Maximum
+	}
+
+	// GKE API provides memory in GB, but ResourceLimiter expects them in bytes
+	if _, found := minLimits[cloudprovider.ResourceNameMemory]; found {
+		minLimits[cloudprovider.ResourceNameMemory] = minLimits[cloudprovider.ResourceNameMemory] * units.Gigabyte
```

Comment:
I know it's just copy-paste, but given our recent problems with MB/MiB I wonder if this is the "correct" gigabyte to use?

Reply:
It's 2^30. Whether this is what we want to do or not is another question :) Either way, it seems to be an API decision and outside the scope of refactoring.